### PR TITLE
Fix bug in CircuitManager

### PR DESCRIPTION
**Pull request by rynowak**
This bug was introduced by removing the parsing from
8fa4df9bda938def3ea053a5c51ed4a38d9b80ec

It looks like these tests are not currently running on the CI which was
how we were able to find the issue.

### CODE REVIEW

File: src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts [hunk 1]
```diff
@@ -91,15 +91,13 @@ function getComponentStartComment(node: Node): StartComponentComment | undefined
...
-            rendererId: rendererId,
-            componentId: componentId,
+            rendererId: Number.parseInt(rendererId),
+            componentId: Number.parseInt(componentId),
           };
-        } else {
-          throw new Error(`Found malformed start component comment at ${node.textContent}`);
         }
       } catch (error) {
-        throw new Error(`Found malformed start component comment at ${node.textContent}`);
```
Comment (rynowak) [inline src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts]: Another minor thing here, the catch block would be hit if like 98 threw. I just moved this to a single throw statement for clarity.

File: src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts [hunk 2]
```diff
@@ -91,15 +91,13 @@ function getComponentStartComment(node: Node): StartComponentComment | undefined
           return {
             node: node as Comment,
             circuitId,
-            rendererId: rendererId,
-            componentId: componentId,
+            rendererId: Number.parseInt(rendererId),
+            componentId: Number.parseInt(componentId),
```
Comment (rynowak) [inline src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts]: This was removed in the previous commit and caused the regression. We *do* have a test for mixed string and int already which is how we found the break.

Comment (javiercn) [reply src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts]: This is weird, it should not be needed.

https://github.com/aspnet/AspNetCore/blob/7b56439ca1cbc384f1e9648f4ae983d3b39119d8/src/Components/Server/src/Circuits/CircuitPrerenderer.cs#L185-L199

Comment (rynowak) [reply src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts]: we have tests for it bruh. That verify that can parse when rendererId/circuitId is a string 😆 

The end-tag version does a conversion as well.

Comment (javiercn) [reply src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts]: Which tests?

Comment (javiercn) [reply src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts]: https://github.com/aspnet/AspNetCore/blob/master/src/Components/Server/test/Circuits/CircuitPrerendererTest.cs#L23

Comment (rynowak) [reply src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts]: I'm happy to fix this in either direction, but something needs to change.

Comment (javiercn) [reply src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts]: If the tests are JS tests, they aren't up to date.

**Comment by rynowak** [comment]
@javiercn updated

**Review by javiercn — APPROVED** [review]

**Comment by SteveSandersonMS** [comment]
> It looks like these tests are not currently running on the CI which was how we were able to find the issue.

Is there anything tracking the need to run these tests in CI?

**Comment by javiercn** [comment]
> Is there anything tracking the need to run these tests in CI?

there is now https://github.com/aspnet/AspNetCore/issues/12231 :)

**Pull request merged**